internal/email/email.go

```diff
@@ -61,56 +111,88 @@ func NewMessage(
+	if err != nil {
+		return "", err
+	}
+	err = bodyTextTmpl.Execute(ctx, &builder, b, BodyMaxLength)
 	if err != nil {
-		zerolog.Ctx(ctx).Error().Err(err).Msg("error parsing the HTML template for email invitations")
-		// Default to the text body
-		return getEmailBodyText(inviteURL, sponsor, project, role)
+		return "", err
 	}
-	// Execute the template
-	var b strings.Builder
-	if err := tmpl.Execute(&b, data); err != nil {
-		return ""
+	return builder.String(), nil
+}
+
+// GetEmailSubject returns the subject for the email based on the message payload
+func (b *bodyData) GetEmailSubject() (string, error) {
+	err := isValidField(b.OrganizationName)
+	if err != nil {
+		return "", err
 	}
-	return b.String()
+	return fmt.Sprintf("You have been invited to join the %s organization in Minder", b.OrganizationName), nil
 }
 
-// getEmailBodyText returns the text body for the email based on the message payload
-func getEmailBodyText(inviteURL, sponsor, project, role string) string {
-	return fmt.Sprintf("You have been invited to join %s as %s by %s. Visit %s to accept the invitation.",
-		project, role, sponsor, inviteURL)
+// isValidField checks if the string is empty or contains HTML or JavaScript injection
+// If we detect any HTML or JavaScript injection, we want to return an error rather than escaping the content
+func isValidField(str string) error {
+	// Check string length
+	if len(str) > MaxFieldLength {
+		return fmt.Errorf("field value %s is more than %d characters", str, MaxFieldLength)
+	}
+	// Check for HTML content
+	escapedHTML := template.HTMLEscapeString(str)
+	if escapedHTML != str {
+		return fmt.Errorf("string %s contains HTML injection", str)
+	}
+
+	// Check for JavaScript content separately
+	escapedJS := template.JSEscaper(str)
+	if escapedJS != str {
+		return fmt.Errorf("string %s contains JavaScript injection", str)
+	}
+
+	return nil
 }
 
-// getEmailSubject returns the subject for the email based on the message payload
-func getEmailSubject(project string) string {
-	return fmt.Sprintf("You have been invited to join the %s organization in Minder", project)
+// Validate validates the template data source for HTML injection attacks
+func (b *bodyData) Validate() error {
+	v := reflect.ValueOf(b).Elem()
+	// Iterate over the fields of the struct
+	for i := 0; i < v.NumField(); i++ {
+		field := v.Field(i)
+		// Check if the field is settable and of kind and type string
+		if field.CanSet() && field.Kind() == reflect.String && field.Type() == reflect.TypeOf("") {
+			strVal := field.String()
+			// Execute your function on the field value
+			err := isValidField(strVal)
+			if err != nil {
+				return fmt.Errorf("field %s failed validation - %s", v.Type().Field(i).Name, strVal)
+			}
+		} else {
+			// Return an error if the field is not settable or not a string
+			return fmt.Errorf("field %s is not settable or a string", v.Type().Field(i).Name)
+		}
```

Comment:
You can avoid some indentation if you reverse this and take advantage of the early-exit of `return`:

```suggestion
		if !field.CanSet() || field.Kind() != reflect.String || field.Type() != reflect.TypeOf("") {
			return fmt.Errorf("field %s is not settable or a string", v.Type().Field(i).Name)
		}
		strVal := field.String()
		err := isValidField(strVal)
		if err != nil {
			return fmt.Errorf("field %s failed validation - %s", v.Type().Field(i).Name, strVal)
		}
```

Reply:
Applied the change 👍